packages/workspace/src/browser/workspace-trust-preferences.ts

```diff
@@ -0,0 +1,84 @@
+/********************************************************************************
+ * Copyright (C) 2021 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { interfaces } from '@theia/core/shared/inversify';
+import {
+    createPreferenceProxy,
+    PreferenceProxy,
+    PreferenceService,
+    PreferenceSchema,
+    PreferenceContribution
+} from '@theia/core/lib/browser/preferences';
+import { nls } from '@theia/core/lib/common/nls';
+import { FrontendApplicationConfigProvider } from '@theia/core/lib/browser/frontend-application-config-provider';
+
+export const WORKSPACE_TRUST_ENABLED = 'security.workspace.trust.enabled';
+export const WORKSPACE_TRUST_STARTUP_PROMPT = 'security.workspace.trust.startupPrompt';
+export const WORKSPACE_TRUST_EMPTY_WINDOW = 'security.workspace.trust.emptyWindow';
+
+export enum WorkspaceTrustPrompt {
+    ALWAYS = 'always',
+    ONCE = 'once',
+    NEVER = 'never'
+}
+
+const CONFIG_AUTO_TRUST = 'autoTrust';
+
+export const workspaceTrustPreferenceSchema: PreferenceSchema = {
+    type: 'object',
+    properties: {
+        [WORKSPACE_TRUST_ENABLED]: {
+            description: nls.localize('theia/workspace/trustEnabled', 'Controls whether or not workspace trust is enabled. If disabled, all workspaces are trusted.'),
+            type: 'boolean',
+            default: FrontendApplicationConfigProvider.get()[CONFIG_AUTO_TRUST] === false
+
+        },
+        [WORKSPACE_TRUST_STARTUP_PROMPT]: {
+            description: nls.localizeByDefault('Controls when the startup prompt to trust a workspace is shown.'),
+            enum: Object.values(WorkspaceTrustPrompt),
+            default: typeof FrontendApplicationConfigProvider.get()[CONFIG_AUTO_TRUST] === 'boolean' ? WorkspaceTrustPrompt.NEVER : WorkspaceTrustPrompt.ALWAYS
```

Comment:
Here the norm would be to implement a default in the `preferences` section of the application configuration rather than introducing a separate field. If you'd like to implement a separate field, you probably want to check the value of the field, not just its type.

Reply:
You are right, we can simply override the preferences in the application configuration so there is no real need for a dedicated field. I changed it to "normal" default values.